src/meilisearch/mod.rs

```diff
@@ -0,0 +1,312 @@
+use std::{borrow::Cow, collections::HashMap};
+
+use testcontainers::{
+    core::{wait::HttpWaitStrategy, ContainerPort, WaitFor},
+    Image,
+};
+
+const NAME: &str = "getmeili/meilisearch";
+const TAG: &str = "v1.8.3";
+const MEILISEARCH_PORT: ContainerPort = ContainerPort::Tcp(7700);
+
+/// Module to work with [`Meilisearch`] inside of tests.
+///
+/// Starts an instance of Meilisearch.
+/// This module is based on the official [`Meilisearch docker image`] documented in the [`Meilisearch docker docs`].
+///
+/// # Example
+/// ```
+/// use testcontainers_modules::{meilisearch, testcontainers::runners::SyncRunner};
+///
+/// let meilisearch_instance = meilisearch::Meilisearch::default().start().unwrap();
+///
+/// let dashboard = format!(
+///     "http://{}:{}",
+///     meilisearch_instance.get_host().unwrap(),
+///     meilisearch_instance.get_host_port_ipv4(7700).unwrap()
+/// );
+/// ```
+///
+/// [`Meilisearch`]: https://www.meilisearch.com/
+/// [`Meilisearch docker docs`]: https://www.meilisearch.com/docs/guides/misc/docker
+/// [`Meilisearch docker image`]: https://hub.docker.com/_/getmeili/meilisearch
+#[derive(Debug)]
+pub struct Meilisearch {
+    env_vars: HashMap<String, String>,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum Environment {
+    Production,
+    Development,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum LogLevel {
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
+    Off,
+}
+
+impl Meilisearch {
+    /// Sets the `MASTER_KEY` for the [`Meilisearch`] instance.
+    /// Default `MASTER_KEY` is `None` if not overridden by this function
+    ///
+    /// See the [official docs for this option](https://www.meilisearch.com/docs/learn/configuration/instance_options#master-key)
+    pub fn with_master_key(mut self, master_key: &str) -> Self {
+        self.env_vars
+            .insert("MEILI_MASTER_KEY".to_owned(), master_key.to_owned());
+        self
+    }
+
+    /// Configures analytics for the [`Meilisearch`] instance.
+    /// Default is `false` if not overridden by this function
+    /// This default differs from the dockerfile as we expect tests not to be good analytics.
+    ///
+    /// See the [official docs for this option](https://www.meilisearch.com/docs/learn/configuration/instance_options#log-level)
+    pub fn with_analytics(mut self, enabled: bool) -> Self {
+        if enabled {
+            self.env_vars.remove("MEILI_NO_ANALYTICS");
+        } else {
+            self.env_vars
+                .insert("MEILI_NO_ANALYTICS".to_owned(), "true".to_owned());
+        }
+        self
+    }
+
+    /// Sets the environment of the [`Meilisearch`] instance.
+    /// Default is [Environment::Development] if not overridden by this function.
+    /// Setting it to [Environment::Production] requires authentication via [Meilisearch::with_master_key]
```

Comment:
We could extend the enum to require the key, like `Environment::Production(key)`, but it might be confusing that there are two ways to set master key though. However I guess it’s just impossible to start production without the key.

Just a note

Reply:
I thought about this.
The reason why I did model it the way I did is that the master key can be set without setting the environment to prod..

(I can change it if you prefer this changed)

Reply:
No-no, that makes sense actually